network/dag/dag.go

```diff
@@ -141,27 +129,42 @@ func (d *dag) Migrate() error {
 				return err
 			}
 		}
+
+		// Migrate headsLegacy to single head
+		// Todo: remove after V6 release => then remove headsShelf
+		headRef, err := writer.Get(stoabs.BytesKey(headRefKey))
+		if err != nil {
+			return err
+		}
+		if headRef == nil {
+			log.Logger().Info("Head not stored in metadata, migrating...")
+			heads := d.headsLegacy(tx)
+			if len(heads) != 0 { // ignore for empty node
+				err = d.setHead(tx, heads[0])
```

Comment:
shouldn't this be the head with the highest LC, instead of the first one?

Reply:
indeed